mitosheet/src/components/endo/celleditor/cellEditorUtils.tsx

```diff
@@ -1,36 +1,72 @@
 // Utilities for the cell editor
 
 import { FunctionDocumentationObject, functionDocumentationObjects } from "../../../data/function_documentation";
-import { ColumnHeader, ColumnID, EditorState, SheetData } from "../../../types";
+import { ColumnID, EditorState, Formula, IndexLabel, MitoSelection, SheetData } from "../../../types";
 import { getDisplayColumnHeader, isPrimitiveColumnHeader, rowIndexToColumnHeaderLevel } from "../../../utils/columnHeaders";
+import { getColumnHeadersInSelection, getIndexLabelsInSelection, getSelectedColumnIDsWithEntireSelectedColumn, isSelectionEntireSelectedColumn } from "../selectionUtils";
 import { getCellDataFromCellIndexes } from "../utils";
 
 
+export const getSelectionFormulaString = (selections: MitoSelection[], sheetData: SheetData, rowIndex: any): string => {
+    // For each of the selections, we turn them into a string that goes into the formula
+    const columnHeadersAndIndexLabels: string[] = []
+
+    // If we have any selectiosn that reference the entire column header
+    const entireSelectedColumns = getSelectedColumnIDsWithEntireSelectedColumn(selections, sheetData);
+    entireSelectedColumns.forEach((columnID) => {
+        const columnHeader = sheetData.columnIDsMap[columnID];
+        const formulaIndexLabel = sheetData.index[rowIndex];
+        columnHeadersAndIndexLabels.push(getDisplayColumnHeader(columnHeader) + getDisplayColumnHeader(formulaIndexLabel));
+    })
+
+
+    selections.forEach(selection => {
+        // Skip these as we handle them above
+        if (isSelectionEntireSelectedColumn(selection)) {
+            return;
+        }
```

Comment:
Why not to just move it here if we're doing the check anyways?

Reply:
Oh... nice.